src/tools/kata-ctl/src/arch/x86_64/mod.rs

```diff
@@ -8,10 +8,48 @@ pub use arch_specific::*;
 
 mod arch_specific {
     use anyhow::Result;
+    use crate::check;
+
+    const PROC_CPUINFO: &str = "/proc/cpuinfo";
+    const CPUINFO_DELIMITER: &str = "\nprocessor";
+    const CPUINFO_FLAGS_TAG: &str = "flags";
+    const CPU_FLAGS_INTEL: &'static [&'static str] = &["lm", "sse4_1", "vmx"];
+    const CPU_ATTRIBS_INTEL: &'static [&'static str] = &["GenuineIntel"];
+
+    // check cpu
+    fn check_cpu() -> Result<()> {
+        println!("INFO: check cpu: x86_64");
+
+        let cpu_info = check::get_single_cpu_info(PROC_CPUINFO, CPUINFO_DELIMITER)?;
+
+        let cpu_flags = match check::get_cpu_flags(&cpu_info, CPUINFO_FLAGS_TAG) {
+            Ok(c) => c,
+            Err(err) => panic!("Error parsing cpu flags, file {:?}, {:?}", PROC_CPUINFO, err),
+        };
```

Comment:
You can replace the `panic!()` with an annotated error using `map_err()`:

```suggestion
        let cpu_flags = check::get_cpu_flags(&cpu_info, CPUINFO_FLAGS_TAG).map_err(|e| anyhow!("Error parsing cpu flags, file {:?}, {:?}", PROC_CPUINFO, e))?;
```